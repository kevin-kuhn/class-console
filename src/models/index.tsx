// PRIMARY RESPONSE
export interface IMeta {
  title: string
  description: string
}

export interface IPrimaryChildren {
  type: string
  title: string
  slug: string
}

export interface IResult {
  title: string
  type: string
  slug: string
  description: string
  children: IPrimaryChildren[] | IContentChildren[]
}

// CONTENT RESPONSE
export interface IVideo {
  provider: string
  link: string
  durationInMinutes: number
}

export interface IPdf {
  title: string
  description: string
  link: string
}

export interface IText {
  title: string
  html: string
}

export interface IAnswer {
  text: string
  isCorrect: boolean
}

export interface IExercice {
  question: string
  answers: IAnswer[]
  correction: string
}

export interface IContentChildren {
  type: string
  data: IVideo | IText | IExercice
}

// FULL RESPONSE
export interface IResponse {
	meta: IMeta
	result: IResult
}