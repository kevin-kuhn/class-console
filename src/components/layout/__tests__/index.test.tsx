import { render } from '../../../test/test-utils'
import Layout from '../index'

import '@testing-library/jest-dom'

import { IMeta } from '../../../models'

describe('Layout', () => {
  it('should be in document', () => {
    const { container } = render(
      <Layout meta={{} as IMeta} studyList={[]}>q
        Teste
      </Layout>
    )

    expect(container).toBeInTheDocument()
  })

  it('should match snapshot', () => {
    const { container } = render(
      <Layout meta={{} as IMeta} studyList={[]}>
        Teste
      </Layout>
    )

    expect(container).toMatchSnapshot()
  })
})
