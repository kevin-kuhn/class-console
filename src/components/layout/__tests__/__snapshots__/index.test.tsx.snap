// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should match snapshot 1`] = `
<div>
  <header
    class="header"
  >
    <span
      class="hamburguer"
    >
      <svg
        fill="none"
        height="32"
        stroke="currentColor"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4 6h16M4 10h16M4 14h16M4 18h16"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
    </span>
    <span>
      <svg
        fill="none"
        height="32"
        stroke="currentColor"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 14l9-5-9-5-9 5 9 5z"
        />
        <path
          d="M12 14l6.16-3.422a12.083 12.083 0 01.665 6.479A11.952 11.952 0 0012 20.055a11.952 11.952 0 00-6.824-2.998 12.078 12.078 0 01.665-6.479L12 14z"
        />
        <path
          d="M12 14l9-5-9-5-9 5 9 5zm0 0l6.16-3.422a12.083 12.083 0 01.665 6.479A11.952 11.952 0 0012 20.055a11.952 11.952 0 00-6.824-2.998 12.078 12.078 0 01.665-6.479L12 14zm-4 6v-7.5l4-2.222"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
    </span>
  </header>
  <section
    class="section"
  >
    <aside
      class="aside close"
    >
      <ul
        class="ul"
      />
    </aside>
    <main
      class="main"
    >
      Teste
    </main>
  </section>
</div>
`;
